<a href="https://colab.research.google.com/github/prasanna2019/pandasEDA/blob/main/EDA_using_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import plotly.express as plt

In [3]:
df = pd.read_csv('https://raw.githubusercontent.com/prasanna2019/pandasEDA/main/TFL%20Bus%20Safety.csv', parse_dates=['Date Of Incident'])
df.head(5)

Year Date Of Incident Route        Operator  Group Name  \
0  2015       2015-01-01     1  London General    Go-Ahead   
1  2015       2015-01-01     4       Metroline   Metroline   
2  2015       2015-01-01     5     East London  Stagecoach   
3  2015       2015-01-01     5     East London  Stagecoach   
4  2015       2015-01-01     6       Metroline   Metroline   

             Bus Garage              Borough  \
0  Garage Not Available            Southwark   
1  Garage Not Available            Islington   
2  Garage Not Available             Havering   
3  Garage Not Available  None London Borough   
4  Garage Not Available          Westminster   

                           Injury Result Description Incident Event Type  \
0                          Injuries treated on scene    Onboard Injuries   
1                          Injuries treated on scene    Onboard Injuries   
2  Taken to Hospital – Reported Serious Injury or...    Onboard Injuries   
3  Taken to Hospital – Reported Serious Injury or...    Onboard Injuries   
4        Reported Minor Injury - Treated at Hospital    Onboard Injuries   

  Victim Category Victims Sex Victims Age  
0       Passenger        Male       Child  
1       Passenger        Male     Unknown  
2       Passenger        Male     Elderly  
3       Passenger        Male     Elderly  
4      Pedestrian      Female     Elderly

# Let us understand the data we have.

In [42]:
df.dtypes

Year                                  int64
Date Of Incident             datetime64[ns]
Route                                object
Operator                             object
Group Name                           object
Bus Garage                           object
Borough                              object
Injury Result Description            object
Incident Event Type                  object
Victim Category                      object
Victims Sex                          object
Victims Age                          object
dtype: object

In [6]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23158 entries, 0 to 23157
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Year                       23158 non-null  int64 
 1   Date Of Incident           23158 non-null  object
 2   Route                      23158 non-null  object
 3   Operator                   23158 non-null  object
 4   Group Name                 23158 non-null  object
 5   Bus Garage                 23158 non-null  object
 6   Borough                    23158 non-null  object
 7   Injury Result Description  23158 non-null  object
 8   Incident Event Type        23158 non-null  object
 9   Victim Category            23158 non-null  object
 10  Victims Sex                23158 non-null  object
 11  Victims Age                23158 non-null  object
dtypes: int64(1), object(11)
memory usage: 2.1+ MB


**Number of incidents happened in each year**

In [4]:
data= df.groupby(['Year']).size().reset_index(name='Incidents')
data.head()


Year  Incidents
0  2015       5715
1  2016       6093
2  2017       6573
3  2018       4777

# Incidents occurring each year
Let's see a line graph of incidents occured per year.

In [5]:
graph= plt.line(data,x="Year", y= "Incidents", title="Year vs Incidents")
graph.update_layout(xaxis=dict(tickmode='linear'))
graph.show()



## As seen from the data above, most incidents occurred in the year 2017


Similarly we can find which operator, borough, victim sex and age has most number of incidents.





In [36]:
type_victim= df.groupby('Victim Category').size()
type_victim.idxmax()


'Passenger'

In [35]:
type_victim.max()

18828

**Maximum accidents occur to Passenger victim type, which are 18828**

In [22]:
df['day']= df['Date Of Incident'].dt.day_name()
incindents_by_day= df.groupby('day').size()
incindents_by_day

day
Friday       2942
Monday       4185
Saturday     3203
Sunday       3516
Thursday     2885
Tuesday      3727
Wednesday    2700
dtype: int64

In [23]:
incindents_by_day.idxmax(), incindents_by_day.max()

('Monday', 4185)

In [24]:
incindents_by_day.idxmin(), incindents_by_day.min()

('Wednesday', 2700)

**As seen, most incidents occur on Monday and least occur on Wednesday**

In [36]:
ages=df.groupby('Victims Age').size().reset_index(name='Count')
ages


Victims Age  Count
0       Adult  10754
1       Child   2181
2     Elderly   2769
3     Unknown   7135
4       Youth    319

In [37]:
fig = plt.pie(ages, names='Victims Age', values='Count', title='Category1 Distribution')
fig.show()

**Adults and unkown are the two age groups most affected in the incidents and Youth are the least affected.**